src/test/java/com/google/devtools/build/lib/standalone/StandaloneSpawnStrategyTest.java

```diff
@@ -205,34 +209,56 @@ private ActionExecutionContext createContext() {
   }
 
   @Test
-  public void testBinFalseYieldsException() throws Exception {
+  public void testBinFalseYieldsException() {
     ExecException e = assertThrows(ExecException.class, () -> run(createSpawn(getFalseCommand())));
     assertWithMessage("got: " + e.getMessage())
-        .that(e.getMessage().startsWith("false failed: error executing command"))
+        .that(e.getMessage().contains("failed: error executing command"))
         .isTrue();
   }
 
   private static String getFalseCommand() {
+    if (OS.getCurrent() == OS.WINDOWS) {
+      // No false command on Windows, with use help.exe as an alternative,
+      // the caveat is that the command will have some output to stdout.
+      // Default exit code of help is 1
+      return "C:/windows/system32/help.exe";
+    }
     return OS.getCurrent() == OS.DARWIN ? "/usr/bin/false" : "/bin/false";
   }
 
   private static String getTrueCommand() {
+    if (OS.getCurrent() == OS.WINDOWS) {
+      // No false command on Windows, with use whoami.exe as an alternative,
+      // the caveat is that the command will have some output to stdout.
+      // Default exit code of help is 0
+      return "C:/windows/system32/whoami.exe";
+    }
     return OS.getCurrent() == OS.DARWIN ? "/usr/bin/true" : "/bin/true";
   }
 
   @Test
   public void testBinEchoPrintsArguments() throws Exception {
-    Spawn spawn = createSpawn("/bin/echo", "Hello,", "world.");
+    Spawn spawn;
+    if (OS.getCurrent() == OS.WINDOWS) {
+      spawn = createSpawn("C:/windows/system32/cmd.exe", "/c", "echo", "Hello,", "world.");
```

Comment:
Are the tests assumed to be running on an arbitrary windows machine or do they operate in some sort of controlled environment (I.e. no need for the real windows installation in c:\windows).
If the tests may be executed on a real machine, I think it's a bit too optimistic to assume that windows is always installed in C:\WINDOWS. You should probably be using `%WinDir%` and `%SystemDrive%` environment variables in order to figure out the actual location.

Reply:
The test mostly runs on Bazel CI, where we can safely assume Windows is installed in C:\Windows. Using environment variable doesn't work when launching a binary through spawn action, the underlying CreateProcess system call won't translate it. So I'll keep C:\Windows, but I did some refactor to make it easier to change the Windows system dir in case someone wants to run this test on a machine with a different Windows system dir.